redis/tests/test_cluster_async.rs

```diff
@@ -175,13 +174,10 @@ async fn test_failover(env: &TestClusterContext, requests: i32, value: i32) {
                         .iter_mut()
                         .map(do_failover)
                         .collect::<stream::FuturesUnordered<_>>()
-                        .fold(
-                            Err(anyhow::anyhow!("None")),
-                            |acc: Result<(), _>, result: Result<(), _>| async move {
-                                acc.or(result)
-                            },
-                        )
+                        .filter_map(|map| future::ready(map.ok()))
```

Comment:
I'm not sure this change is appropriate. If I'm understanding this correctly, it changes the behavior a bit, doesn't it? If I'm not mistaken, with your change, the first success will cause early termination, whereas the original goal of the test was to issue the command on all the nodes. Personally, I'm fine w/ just leaving the override in place here as you had previously.